ui/main/src/app/modules/feed/components/map/map.component.ts

```diff
@@ -0,0 +1,221 @@
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {Map as OpenLayersMap} from 'ol';
+import View from 'ol/View';
+import {Tile as TileLayer, Vector as VectorLayer} from 'ol/layer';
+import {OSM, Vector as VectorSource} from 'ol/source';
+import {fromLonLat} from 'ol/proj';
+import {LightCard, Severity} from '@ofModel/light-card.model';
+import {LightCardsFeedFilterService} from '@ofServices/lightcards/lightcards-feed-filter.service';
+import {Subject} from 'rxjs';
+import {takeUntil} from 'rxjs/operators';
+import WKT from 'ol/format/WKT';
+import Overlay from 'ol/Overlay';
+import {Style, Fill, Stroke, Circle} from 'ol/style';
+import {Attribution, defaults as defaultControls} from 'ol/control';
+import {ConfigService} from '@ofServices/config.service';
+
+let self;
+
+@Component({
+    selector: 'of-map',
+    templateUrl: './map.component.html',
+    styleUrls: ['./map.component.scss']
+})
+export class MapComponent implements OnInit, OnDestroy {
+    private unsubscribe$ = new Subject<void>();
+    private map: OpenLayersMap;
+    private vectorLayer: VectorLayer;
+    lightCard: LightCard;
+
+    constructor(private lightCardsFeedFilterService: LightCardsFeedFilterService,
+                private configService: ConfigService) {
+    }
+
+    ngOnInit() {
+        self = this;
+
+        if (this.configService.getConfigValue('feed.geomap.enableMap', false)) {
+            this.drawMap();
+            this.lightCardsFeedFilterService.getFilteredAndSortedLightCards()
+                .pipe(takeUntil(this.unsubscribe$))
+                .subscribe(cards => {
+                    setTimeout(() => this.updateMap(cards), 500);
+                });
+        }
+    }
+
+    private drawMap() {
+        const overlay = this.getClosePopupOverlay();
+        const attribution = new Attribution({
+            collapsible: true
+        });
+
+        const longitude = this.configService.getConfigValue('feed.geomap.initialLongitude', 0);
+        const latitude = this.configService.getConfigValue('feed.geomap.initialLatitude', 0);
+        const zoom = this.configService.getConfigValue('feed.geomap.initialZoom', 1);
+
+        this.map = new OpenLayersMap({
+            view: new View({
+                center: fromLonLat([longitude, latitude]),
+                zoom: zoom
+            }),
+            layers: [
+                new TileLayer({
+                    source: new OSM()
+                })
+            ],
+            target: 'ol-map',
+            overlays: [overlay],
+            controls: defaultControls({attribution: false}).extend([attribution])
+        });
+
+        this.map.on('singleclick', function(evt) {
+            displayFeatureInfo(evt);
+        });
+
+        function displayFeatureInfo(evt) {
```

Comment:
To be more meaningfull , i would name it displayLightCardIfNecessary

Reply:
Done